graphql.go

```diff
@@ -307,24 +313,65 @@ func (c *Client) runWithJSON(ctx context.Context, req *Request, resp interface{}
 	}
 	c.logf(">> headers: %v", r.Header)
 	r = r.WithContext(ctx)
-	res, err := c.sendRequest(r)
-	if err != nil {
-		return err
-	}
-	defer res.Body.Close()
-	var buf bytes.Buffer
-	if _, err := io.Copy(&buf, res.Body); err != nil {
-		return errors.Wrap(err, "reading body")
-	}
-	c.logf("<< %s", buf.String())
-	if err := json.NewDecoder(&buf).Decode(&gr); err != nil {
-		return errors.Wrap(err, "decoding response")
-	}
-	if len(gr.Errors) > 0 {
-		// return first error
-		return gr.Errors[0]
+
+	return c.executeRequest(gr, r)
+}
+
+func (c *Client) executeRequest(gr *graphResponse, r *http.Request) error {
+	gqlRetryConfig := c.retryConfig
+	var body io.Reader = r.Body
+	tryCount := 0
+	for ; tryCount < gqlRetryConfig.MaxTries; tryCount++ {
+		buf := new(bytes.Buffer)
+		r.Body = ioutil.NopCloser(io.TeeReader(body, buf))
+
+		retryConfig := c.retryConfig
```

Comment:
why do we copy retryConfig twice? (once to `gqlRetryConfig` and once here)

Reply:
I use `gqlRetryConfig` for retrying on error field. 
`retryConfig` is for retrying with status code. I need to make a copy here because I want to check if we retry or not on the http call `sendRequest`. If we retried there, I don't want to retry again on the error field. 

Don't think retries of retries is a good idea.